src/openssl.c

```diff
@@ -3674,6 +3687,10 @@ static int pk_new(lua_State *L) {
 			}
 		}
 
+		BIO_free(bio);
+
+		ud = prepsimple(L, PKEY_CLASS);
```

Comment:
Moving this can still result in a leaked `pub` or `prvt`. I don't think you *can* move this allocation.